cloudinit/sources/DataSourceCloudStack.py

```diff
@@ -328,8 +361,17 @@ def get_vr_address(distro):
     return get_default_gateway()
 
 
+def is_platform_viable() -> bool:
+    product_name = dmi.read_dmi_data("system-product-name")
+    if not product_name:
+        LOG.debug("system-product-name not available in dmi data")
+        return False
```

Comment:
Thank you for this additional and I agree that this messaging will be helpful in the event of platforms without DMI  support.